buildpack.go

```diff
@@ -534,6 +545,33 @@ func (d *DependencyResolver) Resolve(id string, version string) (BuildpackDepend
 	return candidate, nil
 }
 
+func archFromPURL(rawPURL string) (string, error) {
+	if len(strings.TrimSpace(rawPURL)) == 0 {
+		return "amd64", nil
+	}
+
+	purl, err := url.Parse(rawPURL)
+	if err != nil {
+		return "", fmt.Errorf("unable to parse PURL\n%w", err)
+	}
+
+	queryParams := purl.Query()
+	if arch, ok := queryParams["arch"]; ok {
+		return arch[0], nil
+	}
+
+	return "amd64", nil
+}
+
+func archFromSystem() string {
+	archFromEnv, ok := os.LookupEnv("BP_ARCH")
```

Comment:
ok, so this is for testing only, right?
like a escape hatch to fake an arch?

Reply:
It's required to make it testable. I'm not sure if it would be useful in practice. Maybe it could be useful with Macbooks (M1/M2s) because the architecture is a little more fluid there. I don't think it's something we'd promote.

If there's a different way you'd want to do this, i.e. not with env variables, to make it testable. I'd be OK with that too. We could make an interface that provides the arch and then push the interface into the module (that would be the hard part). Then it could use the interface to get the arch and we could have a test implementation and an actual implementation.

Reply:
no that's fine this way; we don't want anyway to overcomplicate something that is bound to be temporary anyway